crates/turbo-tasks-memory/src/memory_backend.rs

```diff
@@ -100,7 +100,7 @@ impl MemoryBackend {
     }
 
     pub fn with_all_cached_tasks(&self, mut func: impl FnMut(TaskId)) {
-        for id in self.task_cache.pin().values() {
+        for id in self.task_cache.clone().into_read_only().values() {
```

Comment:
Do we need to clone if we're just immediately taking a read-only view?

Reply:
`into_read_only` consumes `self`, so we need to clone. However, `with_all_cached_tasks` is a debug helper so it doesn't really matter.